### ISSUE TITLE
MAGIC NUMBERS: ParseConfig()

### ISSUE THREAD

**Issue by henrygab** [drom/wavedrom#365]
<details><summary>See lines 47, 59, 72, 84...</summary><P/>

https://github.com/wavedrom/wavedrom/blob/8118f26283e6ecfd4304bdd58103c9621befc6d2/lib/parse-config.js#L42-L87
</details>

Magic number result in fragile code, and make maintenance much more difficult.

Here, the values `20` and `46` are hard-coded, with no indication of where they came from.

Even after review of the code, and understanding when the values are set(*), it's not clear what values caused these choices.  For example, it appears that:

* `yh0` refers to additional height required for the header, respectively, to allow space for `tick` or `tock` being defined;  The default value is zero, and set to magic number `20` when the value exists in the config.
* `yh1` refers to additional height required for a lane's header text; The default value is zero, and set to magic number `46` when the value exists in the config.
* `yf0` is identical to `yh0`, except relates to the footer.
* `yf1` is identical to `yh1`, except relates to the footer.

### What is the genesis of these two magic numbers?

In other words, what is presumed to be true, that makes these numbers work?  Is it the height of the font (as defined by the skin being used) for these elements?

e.g., is the font used for `tick` and `tock` labels exactly 20 pixels tall?
e.g., is the font used for lane header/footer text exactly 46 pixels tall?

**Comment by Ganesh-AT** [comment]
> e.g., is the font used for `tick` and `tock` labels exactly 20 pixels tall?
> e.g., is the font used for lane header/footer text exactly 46 pixels tall?

An inspection of the rendered SVG using browser dev tools reveals that the tick / tock labels as well as the head / foot text are all 11 pt (14.63 px) in size.

The 20 and 46 numbers are probably in place to give some padding / margins to provide the appearance of a true heading text.

At the risk of digressing a little, just a note that the yh1 / yf1 numbers have to be tweaked if support for multi-line header text (there are a couple of outstanding 'issues' reported here about that) is incorporated. That is a pretty tall task without the getBBox() method. Unfortunately, @drom  removed support for that from the code base some years back.
